src/Sarif/RuleUtilities.cs

```diff
@@ -24,33 +22,22 @@ public static Result BuildResult(ResultLevel level, IAnalysisContext context, Re
                 throw new ArgumentNullException(nameof(arguments));
             }
 
-            string[] messageArguments = arguments;
-
             formatId = RuleUtilities.NormalizeFormatId(context.Rule.Id, formatId);
 
-            string targetPath = context.TargetUri?.LocalPath;
-
-            Result result = new Result();
-
-            result.RuleId = context.Rule.Id;
-
-            if (!string.IsNullOrEmpty(targetPath))
+            Result result = new Result
             {
-                // In the event of an analysis target, we always provide
-                // the local path to this item as the 0th argument
-                messageArguments = new string[arguments.Length + 1];
-                messageArguments[0] = Path.GetFileName(targetPath);
-                arguments.CopyTo(messageArguments, 1);
-            }
+                RuleId = context.Rule.Id,
 
-            result.FormattedRuleMessage = new FormattedRuleMessage()
-            {
-                FormatId = formatId,
-                Arguments = messageArguments
-            };
+                FormattedRuleMessage = new FormattedRuleMessage()
+                {
+                    FormatId = formatId,
+                    Arguments = arguments
+                },
 
-            result.Level = level;
+                Level = level
+            };
 
+            string targetPath = context.TargetUri?.LocalPath;
```

Comment:
Since we no longer use `targetPath` to do argument-munging logic, move its declaration down to its point of use.